### PR TITLE
[Improve][CDC] Optimize split state memory allocation in increment phase

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[CDC] Optimize memory allocation in increment phase

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
https://github.com/apache/seatunnel/blob/5f3c9c36a51d8db7aee45583bc186ba549576e36/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mysql-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/MysqlCDCIT.java#L323

Added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/HybridSplitAssigner.java [hunk 1]
```diff
@@ -146,4 +146,9 @@ public void notifyCheckpointComplete(long checkpointId) {
         snapshotSplitAssigner.notifyCheckpointComplete(checkpointId);
         incrementalSplitAssigner.notifyCheckpointComplete(checkpointId);
     }
+
+    public boolean completedSnapshotPhase(List<TableId> tableIds) {
+        return snapshotSplitAssigner.completedSnapshotPhase(tableIds)
+                & incrementalSplitAssigner.completedSnapshotPhase(tableIds);
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/HybridSplitAssigner.java]: Why not use &&？

Comment (hailin0) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/HybridSplitAssigner.java]: Using `&&` will interrupt execution

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/HybridSplitAssigner.java [hunk 2]
```diff
@@ -146,4 +150,22 @@ public void notifyCheckpointComplete(long checkpointId) {
...
+    SnapshotSplitAssigner<C> getSnapshotSplitAssigner() {
+        return snapshotSplitAssigner;
+    }
+
+    public boolean completedSnapshotPhase(List<TableId> tableIds) {
+        return Arrays.asList(
+                        snapshotSplitAssigner.completedSnapshotPhase(tableIds),
+                        incrementalSplitAssigner.completedSnapshotPhase(tableIds))
+                .stream()
+                .allMatch(Predicate.isEqual(true));
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/HybridSplitAssigner.java:169]: @Carl-Zhou-CN updated

**Comment by hailin0** [comment]
image

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/HybridSplitAssigner.java (hunk 2 above)
Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/source/enumerator/HybridSplitAssigner.java:169]: ok

**Comment by Carl-Zhou-CN** [comment]
good job,It seems that there is a lot of recycling inside

**Comment by hailin0** [comment]
image
image

**Review by Carl-Zhou-CN — DISMISSED** [review]
LGTM

**Comment by Carl-Zhou-CN** [comment]
@hailin0  Please check ci

**Comment by hailin0** [comment]
> image image

@Carl-Zhou-CN here

**Review by Carl-Zhou-CN — APPROVED** [review]
+1

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**